File: zad1.go
package main

import (
	"fmt"    // пакет для форматирования и вывода
	"regexp" // регулярное выражение
)

func main() {
	str := "1001tf11flb10001"                // строка
	regular := regexp.MustCompile("1(0+)1")  // условие поиска
	result := regular.FindAllString(str, -1) // результат
	for _, match := range result {
		fmt.Println(match) // возвращает совпадение с регулярным выражением
	}
}

File: zad1.js
const str = "1001tf11flb10001"; // строка
const regular = new RegExp("1(0+)1", "g"); // условие поиска
const result = str.match(regular); // результат

result.forEach(result => { // поиск совпадения
console.log(result); // вывод
});